packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputDate.tsx

```diff
@@ -47,7 +47,10 @@ function GridFilterInputDate(props: GridFilterInputDateProps) {
 
   const InputProps = {
     ...(applying ? { endAdornment: <GridLoadIcon /> } : {}),
-    max: type === 'datetime-local' ? '9999-12-31T23:59' : '9999-12-31',
+    inputProps: {
+      max: type === 'datetime-local' ? '9999-12-31T23:59' : '9999-12-31',
+      ...inputProps,
```

Comment:
The filter input is based on the `TextField` so users except that the props will be forwarded to the root component.

Here you're spreading the top-level `inputProps` inside `InputProps`. It's like:

`<GridFilterInputDate inputProps={{ min: '...' }} />`
-> `<TextField InputProps={{ inputProps: { min: '...' } }} />` (this PR)
-> `<TextField inputProps={{ min: '...' }} />` (how it should behave)